applications/callflow/src/cf_route_req.erl

```diff
@@ -277,14 +277,37 @@ update_call(Flow, NoMatch, ControllerQ, Call) ->
 %% process
 %% @end
 %%-----------------------------------------------------------------------------
--spec maybe_device_redirected(whapps_call:call()) -> whapps_call:call().
-maybe_device_redirected(Call) ->
-    case whapps_call:custom_channel_var(<<"Redirected-By">>, Call) of
+-spec maybe_call_rerouted(whapps_call:call()) -> whapps_call:call().
+maybe_call_rerouted(Call) ->
+    case get_rerouted_by(Call) of
         'undefined' -> Call;
-        Device ->
-            case cf_util:endpoint_id_by_sip_username(whapps_call:account_db(Call), Device) of
-                {'ok', EndpointId } ->
-                    whapps_call:set_authorization(<<"device">>, EndpointId, Call);
-                {'error', 'not_found'} -> Call
+        ReroutedBy ->
+            ReOptions = [{'capture', [1], 'binary'}],
+            case catch(re:run(ReroutedBy, <<".*sip:(.*)@.*">>, ReOptions)) of
+                {'match', [Match]} -> fix_rerouted_call(Match, Call);
+                _ -> Call
             end
     end.
+
+-spec fix_rerouted_call(api_binary(), whapps_call:call()) -> whapps_call:call().
+fix_rerouted_call(ReroutedBy, Call) ->
+    [Username|_] = binary:split(ReroutedBy, <<"@">>),
+    case cf_util:endpoint_id_by_sip_username(whapps_call:account_db(Call), Username) of
+        {'ok', EndpointId} ->
+            C = whapps_call:set_authorization(<<"device">>, EndpointId, Call),
+            OwnerId = cf_attributes:owner_id(C),
+            whapps_call:set_owner_id(OwnerId, C);
+        {'error', 'not_found'} ->
+            Routines = [fun(C) -> whapps_call:set_authorization(<<"undefined">>, <<"undefined">>, C) end
+                        ,fun(C) -> whapps_call:set_owner_id(<<"undefined">>, C) end
+                        ,fun(C) -> whapps_call:set_account_id(<<"undefined">>, C) end
+                       ],
+            lists:foldl(fun(F, C) -> F(C) end, Call, Routines)
```

Comment:
While you're in here, look at `whapps_call:exec/2` and consider using tuples instead of anonymous functions:

```erlang
whapps_call:exec(
    [{fun whapps_call:set_authorization/3, 'undefined', 'undefined'}
     ,{fun whapps_call:set_owner_id/2, 'undefined}
     ,{fun whapps_call:set_account_id/2, 'undefined'}
    ]
    ,Call
)
```

Something along those lines.